File: app/src/main/java/com/erif/contentloader/helper/timer/Delay.java
package com.erif.contentloader.helper.timer;

import android.os.CountDownTimer;

public class Delay {

    private final int duration;
    private final DelayCallback callback;

    public Delay(int seconds, DelayCallback callback) {
        this.duration = seconds * 1000;
        this.callback = callback;
    }

    public Delay(double seconds, DelayCallback callback) {
        this.duration = (int) (seconds * 1000);
        this.callback = callback;
    }

    public void start() {
        Timer timer = new Timer(duration, 1000, callback);
        timer.start();
    }

    private static class Timer extends CountDownTimer {

        private final DelayCallback callback;

        public Timer(long millisInFuture, long countDownInterval, DelayCallback callback) {
            super(millisInFuture, countDownInterval);
            this.callback = callback;
        }

        @Override
        public void onTick(long l) {

        }

        @Override
        public void onFinish() {
            callback.onFinish();
        }
    }



}

File: app/src/main/java/com/erif/contentloaderdemo/helper/timer/Delay.kt
package com.erif.contentloaderdemo.helper.timer

import android.os.CountDownTimer

class Delay {

    private var duration = 0
    private var callback: DelayCallback? = null

    constructor(seconds: Int, callback: DelayCallback?) {
        duration = seconds * 1000
        this.callback = callback
    }

    constructor(seconds: Double, callback: DelayCallback?) {
        duration = (seconds * 1000).toInt()
        this.callback = callback
    }

    fun start() {
        val timer = Timer(duration.toLong(), 1000, callback)
        timer.start()
    }

    private class Timer(
        millisInFuture: Long,
        countDownInterval: Long,
        private val callback: DelayCallback?
    ) :
        CountDownTimer(millisInFuture, countDownInterval) {
        override fun onTick(l: Long) {}
        override fun onFinish() {
            callback!!.onFinish()
        }
    }

}